File: sample/src/main/java/com/afollestad/aestheticsample/MainPagerAdapter.java
package com.afollestad.aestheticsample;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

/** @author Aidan Follestad (afollestad) */
public class MainPagerAdapter extends FragmentStatePagerAdapter {

  private final Context context;

  public MainPagerAdapter(Context context, FragmentManager fm) {
    super(fm);
    this.context = context;
  }

  @Override
  public Fragment getItem(int position) {
    return position == 0 ? new MainFragment() : new SecondaryFragment();
  }

  @Override
  public int getCount() {
    return 2;
  }

  @Override
  public CharSequence getPageTitle(int position) {
    return context.getString(position == 0 ? R.string.main : R.string.other);
  }
}

File: sample/src/main/java/com/afollestad/aestheticsample/MainPagerAdapter.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.aestheticsample

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/** @author Aidan Follestad (afollestad) */
internal class MainPagerAdapter(
  private val context: Context,
  fm: FragmentManager
) : FragmentStatePagerAdapter(fm) {

  override fun getItem(position: Int): Fragment {
    return if (position == 0) MainFragment() else SecondaryFragment()
  }

  override fun getCount(): Int {
    return 2
  }

  override fun getPageTitle(position: Int): CharSequence? {
    return context.getString(if (position == 0) R.string.main else R.string.other)
  }
}
